apps/aetestframework/lib/worker.ex

```diff
@@ -1,599 +1,188 @@
-      "encoded_block = :erlang.term_to_binary(top_block)"
-    )
-
-    send_command(
-      node_name,
-      "{:respond_top_block, Base.encode32(encoded_block)}"
-    )
-
-    send_command(node_name, " ")
-  end
-
-  @spec update_node_top_block_hash(String.t()) :: :ok | :unknown_node
-  def update_node_top_block_hash(node_name) do
-    send_command(node_name, "block_hash = Chain.top_block_hash() |> :erlang.term_to_binary()")
-    send_command(node_name, "{:respond_hash, Base.encode32(block_hash)}")
-    send_command(node_name, " ")
-  end
-
-  # naming txs
-
   @doc """
-    Functions to pre_claim, claim, update, transfer, revoke naming txs.
-    They have already defined arguments taken from the tests
+  Call a GenServer API function with specific delay
   """
-  @spec naming_pre_claim(String.t()) :: :ok | :unknown_node
-  def naming_pre_claim(node_name) do
-    send_command(
-      node_name,
-      "{:ok, tx} = Account.pre_claim(\"test.aet\", 123, 10)"
-    )
-
-    send_command(node_name, "Pool.add_transaction(tx)")
-  end
-
-  @spec naming_claim(String.t()) :: :ok | :unknown_node
-  def naming_claim(node_name) do
-    send_command(
-      node_name,
-      "{:ok, tx} = Account.claim(\"test.aet\", 123, 10)"
-    )
-
-    send_command(node_name, "Pool.add_transaction(tx)")
-  end
-
-  @spec naming_update(String.t()) :: :ok | :unknown_node
-  def naming_update(node_name) do
-    send_command(
-      node_name,
-      "{:ok, tx} = Account.name_update(\"test.aet\", \"{\\\"test\\\":2}\", 10, 5000, 50) "
-    )
-
-    send_command(node_name, "Pool.add_transaction(tx)")
-  end
-
-  @spec naming_transfer(String.t()) :: :ok | :unknown_node
-  def naming_transfer(node_name) do
-    send_command(node_name, "{transfer_to_pub, transfer_to_priv} = Keys.keypair(:sign)")
-
-    send_command(
-      node_name,
-      "{:ok, transfer} = Account.name_transfer(\"test.aet\", transfer_to_pub, 10)"
-    )
-
-    send_command(node_name, "Pool.add_transaction(transfer)")
+  @spec verify_with_delay(reference(), non_neg_integer()) :: any
+  def verify_with_delay(valid?, 0) do
+    valid?.()
   end
 
-  @spec naming_revoke(String.t()) :: :ok | :unknown_node
-  def naming_revoke(node_name) do
-    send_command(node_name, "{transfer_to_pub, transfer_to_priv} = Keys.keypair(:sign)")
-
-    send_command(
-      node_name,
-      "{:ok, spend} = Account.spend(transfer_to_pub, 15, 10, <<\"payload\">>)"
-    )
-
-    send_command(node_name, "Pool.add_transaction(spend)")
-    mine_sync_block(node_name)
-
-    send_command(
-      node_name,
-      "next_nonce = Account.nonce(Chain.chain_state().accounts, transfer_to_pub) + 1"
-    )
-
-    send_command(
-      node_name,
-      "{:ok, revoke} = Account.name_revoke(transfer_to_pub, transfer_to_priv, \"test.aet\", 10, next_nonce)"
-    )
-
-    send_command(node_name, "Pool.add_transaction(revoke)")
+  def verify_with_delay(valid?, execute_times) do
+    if valid?.() do
+      true
+    else
+      :timer.sleep(1000)
```

Comment:
Maybe lower the sleep to 100ms? Running check more frequently should lower the time it takes for a test to pass.